add troubleshooting ssl cert provisioning in pages (#12030)
github/docs: content/pages/getting-started-with-github-pages/securing-your-github-pages-site-with-https.md

3. Under "{% data variables.product.prodname_pages %}," select **Enforce HTTPS**.
  ![Enforce HTTPS checkbox](/assets/images/help/pages/enforce-https-checkbox.png)

## Troubleshooting certificate provisioning ("Certificate not yet created" error")

> On the Pages page, once your custom domain is saved, an automatic DNS check should begin. If the check is successful, the [Certbot](https://certbot.eff.org/) application running on GitHub's server should automatically request and apply an SSL certificate from [Let's Encrypt](https://letsencrypt.org/). Once this process is complete, you should see a **Certificate Active** status.

Reviewer: Correct. certbot is a CLI that implements the protocol. We use [`acme-client`](https://github.com/unixcharles/acme-client) so we can call it from Ruby code.

> On the Pages page, once your custom domain is saved, an automatic DNS check should begin. If the check is successful, the [Certbot](https://certbot.eff.org/) application running on GitHub's server should automatically request and apply an SSL certificate from [Let's Encrypt](https://letsencrypt.org/). Once this process is complete, you should see a **Certificate Active** status.

Reviewer: Some phrasing suggestions that talk about why we do a DNS check first.

```suggestion
In the Pages settings, once your custom domain is saved, an automatic DNS check should begin. This check determines if your DNS settings are configured to allow us to obtain a certificate automatically. If the check is successful, we queue a job to request a TLS certificate from [Let's Encrypt](https://letsencrypt.org/). Once we have a valid certificate, we automatically upload it to the servers that handle TLS termination for Pages. Once this process is complete, you should see a **Certificate Active** status.
```

> On the Pages page, once your custom domain is saved, an automatic DNS check should begin. If the check is successful, the [Certbot](https://certbot.eff.org/) application running on GitHub's server should automatically request and apply an SSL certificate from [Let's Encrypt](https://letsencrypt.org/). Once this process is complete, you should see a **Certificate Active** status.

Reviewer 2: ```suggestion
When you set or change your custom domain in the Pages settings, an automatic DNS check begins. This check determines if your DNS settings are configured to allow {% data variables.product.prodname_dotcom %} to obtain a certificate automatically. If the check is successful, {% data variables.product.prodname_dotcom %} queues a job to request a TLS certificate from [Let's Encrypt](https://letsencrypt.org/). On receiving receiving a valid certificate, {% data variables.product.prodname_dotcom %} automatically uploads it to the servers that handle TLS termination for Pages. When this process completes successfully, a check mark is displayed beside your custom domain name.
```

> The process may take some time. If, several afters after clicking Save, the process seems to be "stuck" without a successful Certificate Active status, try clicking Remove next to your custom domain name. Retype the domain name and click Save once again. This will cancel and restart the provisioning process.

Reviewer 3: ```suggestion
The process may take some time. If, several minutes after clicking Save, the process seems to be "stuck" without a successful Certificate Active status, try clicking Remove next to your custom domain name. Retype the domain name and click Save once again. This will cancel and restart the provisioning process.
```

> The process may take some time. If, several afters after clicking Save, the process seems to be "stuck" without a successful Certificate Active status, try clicking Remove next to your custom domain name. Retype the domain name and click Save once again. This will cancel and restart the provisioning process.

Reviewer 4: ```suggestion
The process may take some time. If several minutes afters after clicking Save, the process seems to be "stuck" without a successful Certificate Active status, try clicking Remove next to your custom domain name. Retype the domain name and click Save once again. This will cancel and restart the provisioning process.
```

> The process may take some time. If, several afters after clicking Save, the process seems to be "stuck" without a successful Certificate Active status, try clicking Remove next to your custom domain name. Retype the domain name and click Save once again. This will cancel and restart the provisioning process.

Reviewer 2: ```suggestion
The process may take some time. If, several minutes after clicking **Save**, the process has not completed, try clicking **Remove** next to your custom domain name. Retype the domain name and click **Save** again. This will cancel and restart the provisioning process.
```
